typescript.example/examples/minimal-typecheck.ts

```diff
@@ -1,15 +1,18 @@
-var a;
-
-function f(x: string) {
-  return x;
+class Vehicle {
+	wheels: number
 }
 
-f(a);
+class Car extends Vehicle { 
+   engine: string
+} 
+ 
+var c: Car;
+const x = c.wheels;
 
-interface X {}
+interface E {
+	engine: string,
+	wheels: number
+}
+var z: E = c;
```

Comment:
This could be solved by a construct that takes all declarations in one scope and checks whether they are *reachable* in another scope